File: codechat/codechat/backends/mock.py
import time

class MockBackend:
    def chat(self, prompt, stream=True, on_stream=None):
        response = "```python\nprint('Hello from Mock!')\n```"
        if stream:
            for c in response:
                if on_stream:
                    on_stream(c)
                time.sleep(0.001)
        else:
            return response

File: src/backends/mock.ts
import { ModelBackend, StreamCallback } from './base';

export class MockBackend extends ModelBackend {
  async chat(prompt: string, onStream?: StreamCallback): Promise<string | void> {
    const resp = `\`\`\`python
# Mock improved file
print("Hello from Mock")
\`\`\``;
    if (onStream) {
      for (const ch of resp) {
        onStream(ch);
        await new Promise((r) => setTimeout(r, 2));
      }
      return;
    }
    return resp;
  }
}
